.github/workflows/publish_connectors.yml

```diff
@@ -127,156 +144,61 @@ jobs:
         shell: bash
         run: ./poe-tasks/validate-connector-metadata.sh --name ${{ matrix.connector }}
 
-      - name: Publish to Python Registry [On merge to master]
-        id: publish-python-registry-master
-        if: github.event_name == 'push' && steps.connector-metadata.outputs.connector-language == 'python'
-        shell: bash
-        run: |
-          ./poe-tasks/publish-python-registry.sh --name ${{ matrix.connector }}
-        env:
-          PYTHON_REGISTRY_TOKEN: ${{ secrets.PYPI_TOKEN }}
-        continue-on-error: true # We allow this to fail because we are testing this step. TODO: remove this once we are sure it works.
-
-      - name: Publish to Python Registry [Manual]
-        id: publish-python-registry-manual
-        if: (github.event_name == 'workflow_dispatch' || github.event_name == 'workflow_call') && steps.connector-metadata.outputs.connector-language == 'python'
+      - name: Publish to Python Registry
+        id: publish-python-registry
+        if: steps.connector-metadata.outputs.connector-language == 'python'
         shell: bash
         run: |
-          # Determine if this is a pre-release
-          is_pre_release=$(echo "${{ inputs.publish-options }}" | grep -c -- '--pre-release' || true)
-          if [[ $is_pre_release -gt 0 ]]; then
-            PRE_RELEASE_FLAG="--pre-release"
-          else
-            PRE_RELEASE_FLAG=""
-          fi
-
-          ./poe-tasks/publish-python-registry.sh --name ${{ matrix.connector }} $PRE_RELEASE_FLAG
+          ./poe-tasks/publish-python-registry.sh --name ${{ matrix.connector }} --release-type ${{ needs.publish_options.outputs.release-type }}
         env:
           PYTHON_REGISTRY_TOKEN: ${{ secrets.PYPI_TOKEN }}
         continue-on-error: true # We allow this to fail because we are testing this step. TODO: remove this once we are sure it works.
 
-      - name: Upload Python Dependencies to GCS [On merge to master]
+      - name: Upload Python Dependencies to GCS
         id: upload-python-dependencies-master
-        if: github.event_name == 'push' && steps.connector-metadata.outputs.connector-language == 'python'
+        if: steps.connector-metadata.outputs.connector-language == 'python'
         shell: bash
         env:
           GCS_CREDENTIALS: ${{ secrets.METADATA_SERVICE_PROD_GCS_CREDENTIALS }}
         run: |
-          ./poe-tasks/upload-python-dependencies.sh --name ${{ matrix.connector }} --bucket prod-airbyte-cloud-connector-metadata-service
+          ./poe-tasks/upload-python-dependencies.sh \
+            --name ${{ matrix.connector }} \
+            --bucket prod-airbyte-cloud-connector-metadata-service \
+            --release-type ${{ needs.publish_options.outputs.release-type }}
         continue-on-error: true # We allow this to fail because we are testing this step. TODO: remove this once we are sure it works.
```

Comment:
are we good to remove this clause? ... and/or remove this entire step, if we don't actually care about it?